Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1137,6 +1137,95 @@ class BrowserViewController: UIViewController {
         return tabManager.selectedTab?.webView?.becomeFirstResponder() ?? false
     }
 
+    func reloadTab(){
+        if(homePanelController == nil){
+            tabManager.selectedTab?.reload()
+        }
+    }
+    
+    func goBack(){
+        if(tabManager.selectedTab?.canGoBack == true && homePanelController == nil){
+            tabManager.selectedTab?.goBack()
+        }
+    }
+    
+    func findOnPage(){
+        if(homePanelController == nil){
+            tab( (tabManager.selectedTab)!, didSelectFindInPageForSelection: "")
+        }
+    }
+    
+    func openLocation(){
+        urlBar.tabLocationViewDidTapLocation(urlBar.locationView)
+    }
+    
+    func newTab(){
+        openBlankNewTabAndFocus(isPrivate: false)
+    }
+    func newPrivateTab(){
+        openBlankNewTabAndFocus(isPrivate: true)
+    }
+    
+    func closeTab(){
+        if(tabManager.tabs.count > 1){
+            tabManager.removeTab(tabManager.selectedTab!);
+        }
+        else{
+            //need to close the last tab and show the favorites screen thing
+        }
+    }
+    
+    func nextTab(){
+        if(tabManager.selectedIndex < (tabManager.tabs.count - 1) ){
+            tabManager.selectTab(tabManager.tabs[tabManager.selectedIndex+1])
+        }
+        else{
+            if(tabManager.tabs.count > 1){
+                tabManager.selectTab(tabManager.tabs[0]);
+            }
+        }
+    }
+    
+    func previousTab(){
+        if(tabManager.selectedIndex > 0){
+            tabManager.selectTab(tabManager.tabs[tabManager.selectedIndex-1])
+        }
+        else{
+            if(tabManager.tabs.count > 1){
+                tabManager.selectTab(tabManager.tabs[tabManager.count-1])
+            }
+        }
+    }
+
+    override var keyCommands: [UIKeyCommand]? {
+        if #available(iOS 9.0, *) {
+            return [
+                UIKeyCommand(input: "r", modifierFlags: .Command, action: #selector(BrowserViewController.reloadTab), discoverabilityTitle: Strings.ReloadPageTitle),
+                UIKeyCommand(input: "[", modifierFlags: .Command, action: #selector(BrowserViewController.goBack), discoverabilityTitle: Strings.BackTitle),
```

Comment:
Two notes: we need both a `Back` and a `Forward` item. But these cannot be added statically. You have to take a look at the current Tab's and ask it if it is actually possible to go back or forward.